source/common/tracing/zipkin/span_buffer.cc

```diff
@@ -2,7 +2,7 @@
 
 namespace Zipkin {
 
-bool SpanBuffer::addSpan(SpanPtr&& span) {
+bool SpanBuffer::addSpan(Span&& span) {
```

Comment:
OK, be aware though that you are going to run destructors multiple times even if you implement move constructor. I'm OK with this for now as long as the move constructor is correctly implemented and there is a comment somewhere as to why it is necessary. We may want to change the interfaces to use shared_ptr and shared_from_this but I wouldn't do that now. We can see what shows up on perf traces.